Python/Product/VSCode/AnalysisVsc/LanguageServer.cs

```diff
@@ -108,18 +109,32 @@ private void OnUnregisterCapability(object sender, UnregisterCapabilityEventArgs
 
         #region Workspace
         [JsonRpcMethod("workspace/didChangeConfiguration")]
-        public Task DidChangeConfiguration(JToken token) {
-            var @params = token.ToObject<DidChangeConfigurationParams>();
-            if (!(@params.settings is JObject rootSection)) {
-                return Task.CompletedTask;
-            }
+        public async Task DidChangeConfiguration(JToken token) {
+            var settings = new LanguageServerSettings();
 
-            if (!rootSection.TryGetValue("python", out var pythonSection)) {
-                return Task.CompletedTask;
+            var rootSection = token["settings"];
+            var pythonSection = rootSection?["python"];
+            if(pythonSection == null) {
+                return;
             }
 
-            var settings = pythonSection.ToObject<LanguageServerSettings>();
-            return _server.DidChangeConfiguration(new DidChangeConfigurationParams { settings = settings });
+            var autoComplete = pythonSection["autoComplete"];
+            settings.completion.showAdvancedMembers = GetSetting(autoComplete, "showAdvancedMembers", true);
+
+            var analysis = pythonSection["analysis"];
+            settings.analysis.openFilesOnly = GetSetting(analysis, "openFilesOnly", false);
+
+            settings.analysis.errors = GetSetting(analysis, "errors", Array.Empty<string>());
```

Comment:
Are there any problem with converting `JToken` into `LanguageServerSettings` directly?

Reply:
`LanguageServerSettings` isn't an LSP structure (and I've already suggested changing it so that it doesn't look like a JSON object either).

Also, we think there's a 99% change that the settings sent from the editor won't match how we want to use it, at least once PM creates a design for how users will set the settings... so may as well have the code here so we can easily adapt it later.

Reply:
> LanguageServerSettings isn't an LSP structure 

Ok, but why we can't use JSON.Net to feel properties of the object instead of doing it manually? JSON.Net compiles all converters, so there is no performance issue.

Reply:
This is sort of like TS side does it - i.e. fetching settings one by one in most cases (there are couple of cases of direct conversion from interface to object). The code translates extension json settings to the structure we have internally. Fields and their types may or may not match extension json.